src/openrct2/actions/SmallSceneryRemoveAction.cpp

```diff
@@ -132,26 +135,18 @@ GameActions::Result::Ptr SmallSceneryRemoveAction::Execute() const
 
 TileElement* SmallSceneryRemoveAction::FindSceneryElement() const
 {
-    TileElement* tileElement = map_get_first_element_at(_loc);
-    if (!tileElement)
-        return nullptr;
-
-    do
+    const bool isGhost = GetFlags() & GAME_COMMAND_FLAG_GHOST;
+    for (auto* sceneryElement : TileElementsView<SmallSceneryElement>(_loc))
     {
-        if (tileElement->GetType() != TILE_ELEMENT_TYPE_SMALL_SCENERY)
+        if (sceneryElement->IsGhost() != isGhost)
             continue;
-        if ((tileElement->AsSmallScenery()->GetSceneryQuadrant()) != _quadrant)
+        if (sceneryElement->GetSceneryQuadrant() != _quadrant)
             continue;
-        if (tileElement->GetBaseZ() != _loc.z)
+        if (sceneryElement->GetBaseZ() != _loc.z)
             continue;
-        if (tileElement->AsSmallScenery()->GetEntryIndex() != _sceneryType)
+        if (sceneryElement->GetEntryIndex() != _sceneryType)
             continue;
-        if ((GetFlags() & GAME_COMMAND_FLAG_GHOST) && tileElement->IsGhost() == false)
```

Comment:
This is subtly different. Previously, it would accept this action being called with the `GAME_COMMAND_FLAG_GHOST` *not* set and the tile element being a ghost (but not the other way round).

I'm not sure if that was correct or not.

Reply:
You are right, however I believe the change is now correct. If ghost flag is set in the action should only interact with ghost elements. I will do some testing.

Reply:
I think the reason it previously worked was because you only ever have one ghost, this would have probably caused issue if you had more than one.

Reply:
I can't get any difference using the old logic, so it seems fine.